### PR TITLE
fix: Fixing 'access' for charts and report tables [DHIS2-10674]

### PR DESCRIPTION
**Pull request by maikelarabori**
As we have to still support Chart and ReportTable, but behind the scenes, all rules and tables are tight to Visualization, we need to add a "special condition" in order to load the "access" for the Visualization associated with the legacy object returned (Chart or Visualization).
Otherwise, we end up with a different/invalid "access" for Chart and ReportTable.

This PR will ensure the "access" is consistent across all those three entities.
This code must be removed once we completely drop the support for Chart and ReportTable.

### CODE REVIEW

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/visualization/ConversionHelper.java [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+ * @author maikel arabori
+ */
+@Deprecated
+public class ConversionHelper
+{
+    public static List<Chart> convertToChartList( List<Visualization> entities )
+    {
+        final List<Chart> charts = new ArrayList<>();
+
+        if ( isNotEmpty( entities ) )
```
Comment (mortenoh) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/visualization/ConversionHelper.java:61]: Maybe better with !isNotEmpty(..) then return?

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/visualization/ConversionHelper.java [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+        }
+
+        return charts;
+    }
+
+    public static Visualization convertToVisualization( final Chart chart )
+    {
+        final Visualization visualization = new Visualization();
+
+        if ( chart != null )
```
Comment (mortenoh) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/visualization/ConversionHelper.java:119]: Maybe better with if chart == null then return? too many levels on indentation is not good

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/visualization/ConversionHelper.java [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+        }
+
+        return visualization;
+    }
+
+    public static List<ReportTable> convertToReportTableList( List<Visualization> entities )
+    {
+        List<ReportTable> reportTables = new ArrayList<>();
+
+        if ( isNotEmpty( entities ) )
```
Comment (mortenoh) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/visualization/ConversionHelper.java:182]: !isNotEmpty(...)

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/visualization/ConversionHelper.java [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+        }
+
+        return reportTables;
+    }
+
+    public static Visualization convertToVisualization( final ReportTable reportTable )
+    {
+        final Visualization visualization = new Visualization();
+
+        if ( reportTable != null )
```
Comment (mortenoh) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/visualization/ConversionHelper.java:218]: Same above

**Review by mortenoh — CHANGES_REQUESTED** [review]

**Comment by larshelge** [comment]
Just a note, we have decided to remove the `Chart` and `ReportTable` APIs, as well as the classic pivot table app in 2.37. This might still be helpful in case we plan to backport this fix.

**Review by larshelge — APPROVED** [review]

**Pull request merged**